inference-engine/tests/functional/plugin/cpu/single_layer_tests/topk.cpp

```diff
@@ -0,0 +1,349 @@
+                    const auto absoluteDifference = CommonTestUtils::ie_abs(res - ref);
+                    if (absoluteDifference <= threshold) {
+                        continue;
+                    }
+                    double max;
+                    if (sizeof(T_IE) < sizeof(T_NGRAPH)) {
+                        max = std::max(CommonTestUtils::ie_abs(T_NGRAPH(res)), CommonTestUtils::ie_abs(ref));
+                    } else {
+                        max = std::max(CommonTestUtils::ie_abs(res), CommonTestUtils::ie_abs(T_IE(ref)));
+                    }
+                    double diff = static_cast<float>(absoluteDifference) / max;
+                    if (max == 0 || (diff > static_cast<float>(threshold)) ||
+                        std::isnan(static_cast<float>(res)) || std::isnan(static_cast<float>(ref))) {
+                        IE_THROW() << "Relative comparison of values expected: " << ref << " and actual: " << res
+                                   << " at index " << k << " with threshold " << threshold
+                                   << " failed";
+                    }
+                }
+            }
+        }
+    }
+
+private:
+    size_t axis_idx;
+    size_t top_k;
+    size_t outer_size = 1;
+    size_t inner_size = 1;
+    ngraph::opset4::TopK::SortType sort;
+};
+
+TEST_P(TopKLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    Run();
+    CheckPluginRelatedResults(executableNetwork, "TopK");
+}
+
+namespace {
+
+std::vector<CPUSpecificParams> filterCPUInfoForDevice() {
+    std::vector<CPUSpecificParams> resCPUParams;
+    if (with_cpu_x86_avx512f()) {
+        resCPUParams.push_back(CPUSpecificParams{{nchw, x}, {nchw, nchw}, {"jit_avx512"}, "jit_avx512_FP32"});
+        resCPUParams.push_back(CPUSpecificParams{{nhwc, x}, {nhwc, nhwc}, {"jit_avx512"}, "jit_avx512_FP32"});
+        resCPUParams.push_back(CPUSpecificParams{{nChw16c, x}, {nChw16c, nChw16c}, {"jit_avx512"}, "jit_avx512_FP32"});
+    } else if (with_cpu_x86_avx2()) {
+        resCPUParams.push_back(CPUSpecificParams{{nchw, x}, {nchw, nchw}, {"jit_avx2"}, "jit_avx2_FP32"});
+        resCPUParams.push_back(CPUSpecificParams{{nhwc, x}, {nhwc, nhwc}, {"jit_avx2"}, "jit_avx2_FP32"});
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, x}, {nChw8c, nChw8c}, {"jit_avx2"}, "jit_avx2_FP32"});
+    } else if (with_cpu_x86_sse42()) {
+        resCPUParams.push_back(CPUSpecificParams{{nchw, x}, {nchw, nchw}, {"jit_sse42"}, "jit_sse42_FP32"});
+        resCPUParams.push_back(CPUSpecificParams{{nhwc, x}, {nhwc, nhwc}, {"jit_sse42"}, "jit_sse42_FP32"});
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, x}, {nChw8c, nChw8c}, {"jit_sse42"}, "jit_sse42_FP32"});
+    } else {
+        resCPUParams.push_back(CPUSpecificParams{{nchw, x}, {nchw, nchw}, {"ref"}, "ref_FP32"});
+    }
+    return resCPUParams;
+}
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
```

Comment:
Could you please extend the tests for BF16 precision as well.
Both enforcing BF16 for FP32 network and just plain BF16 network options should be verified.
See roi_pooling test as an example.

Reply:
Fixed. Thansk Egor!